pkg/storegateway/indexheader/lazy_binary_reader.go

```diff
@@ -82,6 +82,10 @@ type LazyBinaryReader struct {
 
 	// Keep track of the last time it was used.
 	usedAt *atomic.Int64
+
+	// tracks which blocks are lazy loaded
+	headersLazyLoadedTracker *HeadersLazyLoadedTracker
+	blockId                  string
```

Comment:
These seem unused now, can be removed maybe?

Reply:
It is still used. We need reference to blockid from LazyBinaryReader.